packages/grid/x-data-grid/src/hooks/utils/useGridRootProps.ts

```diff
@@ -2,7 +2,7 @@ import * as React from 'react';
 import { GridRootPropsContext } from '../../context/GridRootPropsContext';
 import type { DataGridProcessedProps } from '../../models/props/DataGridProps';
 
-export const useGridRootProps = () => {
+export const useGridRootProps = <T extends DataGridProcessedProps = DataGridProcessedProps>() => {
```

Comment:
Is the purpose of this change to make it easier to override the utility return type for the `Pro` and `Premium` packages?
We are doing that right now like this: https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid-pro/src/hooks/utils/useGridRootProps.ts#L4
The approach you followed would make the return type of `useGridRootProps` customizable by users which is not the correct thing and may portray wrong information as this hook is populated internally.

Would you happen to see any other benefit of your approach over the one already being followed?

Reply:
We have made some changes to the approach @MBilalShafi, please let us know if this works.